sphinx_gallery/directives.py

```diff
@@ -77,3 +82,200 @@ def run(self):
         self.state_machine.insert_input(include_lines, path)
 
         return []
+
+
+"""
+Image sg for responsive images
+"""
+
+
+class imgsgnode(nodes.General, nodes.Element):
+    pass
+
+
+def directive_boolean(value):
```

Comment:
Similar for this function, it does not seem to be used? Thanks!
cc @jklymak @larsoner

Reply:
I would try removing them and see if anything breaks. We've been decent at adding new tests for new functionality for the last few years so hopefully we can rely on them